### PR TITLE
provider/aws: Fix reading auto scaling group termination policies

### PR DESCRIPTION
**Pull request by tpounds**
Similar fix to #5044 and #5045 that fixes reading an auto scaling groups' termination policies. I also added an acceptance test to make sure the default termination policy is applied as expected.

### CODE REVIEW

**Comment by tpounds** [comment]
FYI. There is a subtle problem with the pull request. I'm in the process of fixing it. I will provide an upate once the issue is fixed.

**Comment by catsby** [comment]
Thanks for letting us know, just ping here when it's ready :smile:

**Comment by tpounds** [comment]
@catsby Should be good to go now. Let me know if you see anything that can be improved.

File: builtin/providers/aws/resource_aws_autoscaling_group.go [hunk 1]
```diff
@@ -251,7 +251,16 @@ func resourceAwsAutoscalingGroupRead(d *schema.ResourceData, meta interface{}) e
...
 	d.Set("tag", g.Tags)
 	d.Set("vpc_zone_identifier", strings.Split(*g.VPCZoneIdentifier, ","))
-	d.Set("termination_policies", g.TerminationPolicies)
+
+	// If no termination polices are explicitly configured and the upstream state
+	// is only using the "Default" policy, clear the state to make it consistent
+	// with the default AWS create API behavior.
+	_, ok := d.GetOk("termination_policies")
+	if !ok && len(g.TerminationPolicies) == 1 && *g.TerminationPolicies[0] == "Default" {
+		d.Set("termination_policies", []string{})
```
Comment (phinze) [inline builtin/providers/aws/resource_aws_autoscaling_group.go]: I believe this should be `[]interface{}`, same as the return type of `flattenStringList`.

Comment (phinze) [inline builtin/providers/aws/resource_aws_autoscaling_group.go]: As a rule of thumb we check the `error` return on non-primitive `d.Set()` calls, which I believe would expose the above issue.

**Comment by phinze** [comment]
Thanks for this, @tpounds! See inline for a few comments.

File: builtin/providers/aws/resource_aws_autoscaling_group.go (hunk 1 above)
Comment (tpounds) [inline builtin/providers/aws/resource_aws_autoscaling_group.go]: Interesting. AFAICT, this doesn't seem like this is done much in practice. I'll fix up to use []interface{} though.

**Comment by tpounds** [comment]
@catsby Updated to use []interface{}. LMK if I should address anything else.

**Comment by phinze** [comment]
@tpounds Sorry didn't make it clear in the comment - `[]interface{}{}` would be the literal declaration of an empty `[]interface{}`.

**Comment by phinze** [comment]
@tamsky Once that is fixed - this looks good to go! Exercised locally and it seems to be working :+1:

**Comment by phinze** [comment]
Whoops! Bad auto-complete there, sorry! Meant @tpounds :stuck_out_tongue_closed_eyes:

**Comment by tpounds** [comment]
Gah! 100% my fault. Should have built locally before pushing. Should be good now. :wink:

**Pull request merged**